### PR TITLE
Neondb branching

### PR DESCRIPTION
**Pull request by Myestery**
/claim #70

### CODE REVIEW

**Comment by dray92** [comment]
@Myestery can you check the lint errors?

**Comment by Myestery** [comment]
> @Myestery can you check the lint errors?

Okay

**Comment by Myestery** [comment]
All good @dray92

File: how-to-guides/third-party/neon-db.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+description: >-
+  Using a NeonDB database branch from a DevZero workspace is a very good use case for workspaces on dev zero.
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:3]:
```suggestion
  Using a NeonDB database branch from a DevZero workspace.
```

File: how-to-guides/third-party/neon-db.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+### Building a Todo App
+
+For the purpose of this guide, we will be using a [todo app](https://github.com/myestery/todo-app) built with nextjs that requires an environment variable called `DATABASE_URL` . We will be creating a recipe for this application and we will launch a workspace based on it.
+
+#### Build Steps
+
+- install linux helper packages
+- clone repo
+- install nodejs
+- run npm install
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:35]:
```suggestion
- run `npm install`
```

File: how-to-guides/third-party/neon-db.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    - type: command
+      command: "echo 'Hello, Build step!'"
+    #
+    ## Here is where you can add code repositories to be cloned into your workspace. Import multiple repositories by adding additional git-clone steps
+    - type: git-clone
+      url: https://github.com/Myestery/todo-app
+    - type: command
+      name: |
+        Download and install NVM, Rationale: File: package.json
+        Install Node.js 21.0.0, Rationale: No Node.js version specified; using default
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:80]:
```suggestion
        Download and install NVM; then install node.js 21
```

File: how-to-guides/third-party/neon-db.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        nvm install 21.0.0
+      directory: /home/devzero
+      user: devzero
+    - type: command
+      name: |
+        Install dependencies
+      command: |
+        . /etc/profile.d/nvm-installation.sh
+        npm install
+      directory: todo-app
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:95]:
```suggestion
        npm install
        npm run build
      directory: todo-app
```

File: how-to-guides/third-party/neon-db.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      user: devzero
+    - type: command
+      name: |
+        Install dependencies
+      command: |
+        . /etc/profile.d/nvm-installation.sh
+        npm run build
+      directory: todo-app
+      user: devzero
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:104]:
```suggestion
```

Comment (dray92) [reply how-to-guides/third-party/neon-db.md:104]: merged into previous section to make it concise

File: how-to-guides/third-party/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# Third Party
```
Comment (dray92) [inline how-to-guides/third-party/README.md:1]: can you add a table here like these ones? https://github.com/devzero-inc/product-docs/blob/staging/how-to-guides/cloud-services/README.md?plain=1

**Comment by Myestery** [comment]
all done @dray92

File: how-to-guides/third-party/neon-db.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+For the purpose of this guide, we will be using a [todo app](https://github.com/myestery/todo-app) built with nextjs that requires an environment variable called `DATABASE_URL` . We will be creating a recipe for this application and we will launch a workspace based on it.
+
+#### Build Steps
+
+- install linux helper packages
+- clone repo
+- install nodejs
+- run `npm install`
+- run `npm run build`: In, this step, we need the database URL to be present so we can run some [prisma](https://www.prisma.io/) commands that are required for manipulating the database
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:36]:
```suggestion
- run `npm run build`: in this step, we need the database URL to be present so we can run some [prisma](https://www.prisma.io/) commands that are required for manipulating the database
```

File: how-to-guides/third-party/neon-db.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+- install linux helper packages
+- clone repo
+- install nodejs
+- run `npm install`
+- run `npm run build`: in this step, we need the database URL to be present so we can run some [prisma](https://www.prisma.io/) commands that are required for manipulating the database
+
+#### Launch Steps
+
+- npm start: to start the application
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:40]:
```suggestion
- `npm start`: to start the application
```

File: how-to-guides/third-party/neon-db.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+          "tar",
+          "unzip",
+          "vim",
+          "wget",
+          "zip",
+        ]
+    - type: command
+      command: "echo 'Hello, Build step!'"
+    #
+    ## Here is where you can add code repositories to be cloned into your workspace. Import multiple repositories by adding additional git-clone steps
```
Comment (dray92) [inline how-to-guides/third-party/neon-db.md:74]:
```suggestion
    # Here is where you can add code repositories to be cloned into your workspace. Import multiple repositories by adding additional git-clone steps
```

**Comment by Sid-Lais** [comment]
1. We could use an uncropped screenshot as we do in our other docs for the architecture.
2. Used a different format to link images and videos.

Used here is:
```
<figure><img src="../../.gitbook/assets/neondb.drawio.png" alt=""><figcaption></figcaption></figure>
```
We used:
```
![Azure VM Architecture](../../../.gitbook/assets/azure-vm-architecture.png)
```

**Comment by Sid-Lais** [comment]
I was talking about the Create a new branch screenshot. For diagrams we use Lucid app

**Comment by Sid-Lais** [comment]
In line 96 I guess launch command should be npm start which you mentioned earlier
The video just goes through the things and does not explain them clearly. Most importantly, the video voice is very low. You could try the app in the end by running some commands or something to show it works.
Should install postgres cli to check the postgres connectivity

**Comment by Sid-Lais** [comment]
> > I was talking about the Create a new branch screenshot. For diagrams we use Lucid app
> 
> Fixed now with a better screenshot.

Delete the prev one

**Comment by dray92** [comment]
@Myestery can you take a look?
rest of the PR seems good and we're ready to merge after this

**Comment by Sid-Lais** [comment]
@Myestery can you check above as I have had put some comments that needs to be resolved

**Comment by Sid-Lais** [comment]
> > @Myestery can you check above as I have had put some comments that needs to be resolved
> 
> I have resolved those ones too

I meant the minor grammatical errors

**Comment by Sid-Lais** [comment]
in how-to-guides/third-party/neon-db.md
1. line 19 It will be Current point in time not to time and make it bold
2. line 124 in generating not ingenerating

**Pull request merged**